pyhive/tests/test_sasl_compat.py

```diff
@@ -0,0 +1,307 @@
+import unittest
+import base64
+import hashlib
+import hmac
+import kerberos
+from mock import patch
+import six
+import struct
+from puresasl import SASLProtocolException, QOP
+from puresasl.client import SASLError
+from pyhive.sasl_compat import PureSASLClient, error_catcher
+
+
+class TestPureSASLClient(unittest.TestCase):

```

Comment:
qq, are those tests moved from somewhere are written from scratch.
If those are moved, let's reference the source as well

Reply:
I took reference of test cases from the pure-sasl repo and refactored them to cover sasl_compat.py and added coverage for functions exclusive to sasl_compat.py.

Reply:
added reference at the top of the file